### ISSUE TITLE
Scan on Webdav root directory with many files causes "stack size exceeded" error

### ISSUE THREAD

**Issue by e-alfred** [icidasset/diffuse#112]
If I try to scan a root directory of a Nextcloud instance over webdav with a lot of files/folders I get the following error:

```
Uncaught RangeError: Maximum call stack size exceeded
slave.js:2897 Uncaught RangeError: Maximum call stack size exceeded
    at Function.indexes [as func] (slave.js:2897)
    at A2 (slave.js:92)
    at _eeue56$elm_xml$Xml_Decode$actualDecode (slave.js:9242)
    at _eeue56$elm_xml$Xml_Decode$decode (slave.js:9238)
    at Function.func (slave.js:9206)
    at A2 (slave.js:92)
    at Function.func (slave.js:2145)
    at A3 (slave.js:98)
    at _eeue56$elm_xml$Xml_Decode$decodeChildren (slave.js:9198)
    at Function.func (slave.js:9184)
```

Aside from the above message. the "processing tracks" message is shown indefinitely.

**Comment by icidasset** [comment]
Thanks! I made a small change to how a WebDav source is processed, could you check if it works for you? If not, can you give me a bit more info on your WebDav server and the amount of files/directories you have?

**Comment by icidasset** [comment]
@e-alfred This should be fixed, please re-open if it happens again. Thanks ;)

**Comment by e-alfred** [comment]
I got the same message with Beta 7. My Webdav server is a Nextcloud 13 instance with around 45 folders at the root and around 100 000 files.

**Comment by icidasset** [comment]
Issue: https://github.com/elm/compiler/issues/1521
Elm seems to have trouble with large lists, this will be fixed in the next version of Elm.

In the meantime, I've decided to parse the XML manually, this is most likely less memory-heavy.
I hope this will fix it 🤞As I can't reproduce this issue myself, sadly.

I've also set `Depth` to `1`, this probably didn't change anything, but just in case.

**Just released Beta 10.**

**Comment by icidasset** [comment]
@e-alfred ☝️Updated.

**Comment by icidasset** [comment]
@e-alfred Ok, I finally found a way to properly test this.

Tested with (1):
- 85GB of music, 13_000 files
- A directory-tree depth of 2
- https://github.com/hacdias/webdav

(2)
- 150 files
- A directory-tree depth of 6
- NextCloud on OwnCloud VirtualBox instance

I adjusted the WebDAV parser as a result of this testing.

If it still gives you this problem, just restart the app.
The processing is done in batches, so it saves the results every now and then.
In other words, it doesn't start from scratch every time there's an error.

Hopefully, when the new version of Elm comes out, this should occur even less.

**Release tag: 1.0.0-beta.14**

PS. I can't believe you have 100_000 files on your instance, that is a LOT of music 😄
